packages/synapse-react-client/src/components/EvaluationFinder/EvaluationFinder.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react'
+import { useGetEvaluationsInfinite } from '../../synapse-queries/evaluation/useEvaluation'
+import { HelpPopover } from '../HelpPopover/HelpPopover'
+import {
+  Box,
+  Button,
+  FormControl,
+  FormGroup,
+  LinearProgress,
+  Typography,
+} from '@mui/material'
+import { GetEvaluationParameters } from '@sage-bionetworks/synapse-types'
+import { Checkbox } from '../widgets/Checkbox'
+
+export type EvaluationFinderProps = Pick<
+  GetEvaluationParameters,
+  'accessType' | 'activeOnly'
+> & {
+  selectedIds: string[]
+  onChange: (newSelectedIds: string[]) => void
+}
+
+export default function EvaluationFinder(props: EvaluationFinderProps) {
+  const { accessType, activeOnly, selectedIds = [], onChange } = props
+  const [currentPage, setCurrentPage] = useState(0)
+  const { data, isLoading, hasNextPage, fetchNextPage, isFetchingNextPage } =
+    useGetEvaluationsInfinite(
+      {
+        accessType,
+        activeOnly,
+      },
+      { keepPreviousData: true },
+    )
+
+  if (isLoading) {
+    return <LinearProgress />
+  }
+  if (!data?.pages) {
+    return <></>
+  }
+
+  const canGoToNextPage =
+    (data.pages.length - 1 > currentPage || hasNextPage) && !isFetchingNextPage
```

Comment:
This suggests that there is a case when `hasNextPage `is `false `but `data.pages.length - 1 > currentPage` is `true`?

Reply:
To be more clear, why is this not enough?  `hasNextPage && !isFetchingNextPage`

Reply:
Mismatch between the infinite query API and how it's used in this component. `hasNextPage` tells us if there's another page on the server. This breaks down with the following sequence:
1. There are two pages of data
2. Component is rendered, page 1 loads
3. Go to page 2. `hasNextPage` is now false
4. Go back to page 1. `hasNextPage` is still false, but we need a way to go to page 2. 

I think I could do a better job of organizing + using better variable names to make this more clear.